### PR TITLE
feat(spanner): add support for change streams transaction exclusion option

### PR DESCRIPTION
**Pull request by ShuranZhang**
Add support for excluding transactions from being recorded in the change streams by passing a new boolean option `ExcludeTxnFromChangeStreams` in the various write APIs:

- Apply
- ApplyAtLeastOnce
- ReadWriteTransactionWithOptions
- BtachWriteWithOptions
- PartitionedUpdateWithOptions

This pr follows the exact same design with the merged Java client lib pr for this feature at https://github.com/googleapis/java-spanner/pull/2959 including error handling.
Passing this option to a request level DML request will receive an error `errExcludeRequestLevelDmlFromChangeStreams`. This new option is added to `struct QueryOptions`, but it can only be used for `PartitionedUpdate` API. 

Note: Samples will be added later in separate prs.

### CODE REVIEW

File: spanner/transaction.go [hunk 1]
```diff
@@ -115,15 +115,20 @@ type TransactionOptions struct {
...
 // order of precedence.
 func (to TransactionOptions) merge(opts TransactionOptions) TransactionOptions {
 	merged := TransactionOptions{
-		CommitOptions:  to.CommitOptions.merge(opts.CommitOptions),
-		TransactionTag: to.TransactionTag,
-		CommitPriority: to.CommitPriority,
+		CommitOptions:               to.CommitOptions.merge(opts.CommitOptions),
+		TransactionTag:              to.TransactionTag,
+		CommitPriority:              to.CommitPriority,
+		ExcludeTxnFromChangeStreams: to.ExcludeTxnFromChangeStreams,
```
Comment (dedocibula) [inline spanner/transaction.go:131]:
```
ExcludeTxnFromChangeStreams: to.ExcludeTxnFromChangeStreams || opts.ExcludeTxnFromChangeStreams
```

Might be cleaner. Same below

File: spanner/pdml_test.go [hunk 2]
```diff
@@ -179,3 +179,25 @@ func TestPartitionedUpdate_Tagging(t *testing.T) {
...
+	}
+	requests := drainRequestsFromServer(server.TestSpanner)
+	if err := compareRequests([]interface{}{
+		&sppb.BatchCreateSessionsRequest{},
+		&sppb.BeginTransactionRequest{},
+		&sppb.ExecuteSqlRequest{}}, requests); err != nil {
+		t.Fatal(err)
+	}
+
+	if !requests[1].(*sppb.BeginTransactionRequest).GetOptions().GetExcludeTxnFromChangeStreams() {
```
Comment (dedocibula) [inline spanner/pdml_test.go:200]: Don't use Get syntax, RequestOptions.ExcluteTxnFromChangeStreams is fine. Same below

Comment (ShuranZhang) [reply spanner/pdml_test.go:200]: Done.

File: spanner/transaction.go (hunk 1 above)
Comment (ShuranZhang) [reply spanner/transaction.go]: Done.

File: spanner/client.go [hunk 3]
```diff
@@ -683,6 +683,10 @@ type applyOption struct {
 	transactionTag string
 	// priority is the RPC priority that is used for the commit operation.
 	priority sppb.RequestOptions_Priority
+	// If excludeTxnFromChangeStreams == true, mutations from this Client.Apply
+	// will not be recorded in allowed tracking change treams with DDL option
```
Comment (rahul2393) [inline spanner/client.go:687]:
```suggestion
	// will not be recorded in allowed tracking change streams with DDL option
```

**Review by rahul2393 — APPROVED** [review]

**Pull request merged**